scalafix-interfaces/src/main/java/scalafix/interfaces/ScalafixPatch.java

```diff
@@ -0,0 +1,9 @@
+package scalafix.interfaces;
+
+public interface ScalafixPatch {
```

Comment:
do we need to expose a patch as single unit? it's coupled to a file and usually use together with others, so instead I would take a kind or an id as parameter on the `apply` overloads (since I guess the only purpose of getting them is to filter before applying them?)

Reply:
discussed offline: this PR is driven by a need to try to apply some patches until files reach a certain state (via `previewSelectivePatches`), but we don't know beforehand which patch will be tried, so we need to be able to address them individually as we try and iterate.